File: ProjecIntegration.Api/ProjecIntegration.Api/Models/Entity/Representation.cs
﻿using ProjecIntegration.Api.Models.BaseEntity;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace ProjecIntegration
{
    public class Representation :BaseEntity
    {
        public int Id { get; set; }
        public string Name { get; set; }
        public string Description { get; set; }
        public string Type { get; set; }
        public DateTime Seance { get; set; }
        public Catalogue Catalogue { get; set; }
        public int CatalogueId { get; set; }
        public SalleDeTheatre SalleDeTheatre { get; set; }
        public int IdSalledeTheatre { get; set; }
    }
}

File: ProjecIntegration.Api/projectintegrationFront/src/app/class/representation.ts
import { SalledeTheatre } from './salledetheatre';
import { Catalogue } from "./catalogue";

export class Representation {
   Name: string;
  Description: string;
  Type: string;
  Seance: Date;
  Catalogue: Catalogue;
  CatalogueId: number;
  SalleDeTheatre: SalledeTheatre;
  IdSalledeTheatre: number;

  constructor(
    Name: string,
    Description: string,
    Type: string,
    Seance: Date,
    Catalogue: Catalogue,
    CatalogueId: number,
    SalleDeTheatre: SalledeTheatre,
    IdSalledeTheatre: number
  ) {
    this.Name = Name;
    this.Description = Description;
    this.Type = Type;
    this.Seance = Seance;
    this.Catalogue = Catalogue;
    this.CatalogueId = CatalogueId;
    this.SalleDeTheatre = SalleDeTheatre;
    this.IdSalledeTheatre = IdSalledeTheatre;
  }
}


